File: Powerful_Integers/Powerful_Integers.py
class Solution:
	def powerfulIntegers(self, x, y, bound):
		res = set()
		i = 0
		while x**i <= bound:
			j = 0
			while y**j <= bound:
				if x**i + y**j <= bound:
					res.add(x**i + y**j)
				if y == 1 and j == 0:
					break
				j += 1
			if x == 1 and i == 0:
				break
			i += 1
		return list(res)

c = Solution()
print(c.powerfulIntegers(2, 1, 10))

File: Powerful_Integers/Powerful_Integers.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int	power(int base, int times)
{
	int res = 1;

	while (times-- > 0)
		res *= base;
	return (res);
}

int* powerfulIntegers(int x, int y, int bound, int* returnSize){
	int	*arr;
	int idx = 0;
	int counter = 0;

	arr = (int *)malloc(sizeof(int) * (bound + 1));
	for (int i = 0; i < bound + 1; i++)
		arr[i] = 0;
	for (int i = 0; power(x, i) <= bound; i++)
	{
		for (int j = 0; power(y, j) <= bound; j++)
		{
			if (power(x, i) + power(y, j) <= bound && !arr[power(x, i) + power(y, j)])
			{
				arr[power(x, i) + power(y, j)] = 1;
				counter++;
			}
		}
	}
	returnSize = (int *)malloc(sizeof(int) * counter);
	for (int i = 0; i < bound + 1; i++)
		if(arr[i])
			returnSize[idx++] = i;
	// for (int i = 0; i < counter; i++)
	// 	printf("%d ", returnSize[i]);
	return (returnSize);
}

int main()
{
	int *res = NULL;
	powerfulIntegers(2, 3, 10, res);

	return (0);
}
